wideq/core.py

```diff
@@ -169,68 +196,146 @@ def __init__(self, device_id, code):
-    authenticated session.
-    """
+        headers["x-emp-token"] = access_token
+
+    if user_number:
+        headers["x-user-no"] = user_number
+
+    return {**headers, **extra_headers}
+
+
+def thinq2_get(
+    url,
+    access_token=None,
+    user_number=None,
+    headers={},
+    country=DEFAULT_COUNTRY,
+    language=DEFAULT_LANGUAGE,
+):
+    """Make an HTTP request in the format used by the API2 servers."""
+
+    # this code to avoid ssl error 'dh key too small'
+    requests.packages.urllib3.disable_warnings()
+    requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS += "HIGH:!DH:!aNULL"
+    try:
+        requests.packages.urllib3.contrib.pyopenssl.DEFAULT_SSL_CIPHER_LIST += (
+            "HIGH:!DH:!aNULL"
+        )
+    except AttributeError:
+        # no pyopenssl support used / needed / available
+        pass
+    # this code to avoid ssl error 'dh key too small'
+
+    LOGGER.debug("thinq2_get before: %s", url)
+
+    res = requests.get(
+        url,
+        headers=thinq2_headers(
+            access_token=access_token,
+            user_number=user_number,
+            extra_headers=headers,
+            country=country,
+            language=language,
+        ),
+        timeout=DEFAULT_TIMEOUT,
+    )
+
+    out = res.json()
+    LOGGER.debug("thinq2_get after: %s", out)
+
+    if "resultCode" not in out:
+        raise APIError("-1", out)
+
+    code = out["resultCode"]
+    if code != "0000":
+        if code in API2_ERRORS:
+            raise API2_ERRORS[code]()
+        raise APIError(code, "error")
+    return out["result"]
+
+
+def lgedm2_post(
+    url,
+    data=None,
+    access_token=None,
+    user_number=None,
+    headers={},
+    country=DEFAULT_COUNTRY,
+    language=DEFAULT_LANGUAGE,
+    use_tlsv1=True,
+):
+    """Make an HTTP request in the format used by the API servers."""
+
+    s = requests.Session()
+    if use_tlsv1:
+        s.mount(url, Tlsv1HttpAdapter())
+    res = s.post(
+        url,
+        json={DATA_ROOT: data},
+        headers=thinq2_headers(
+            access_token=access_token,
+            user_number=user_number,
+            extra_headers=headers,
+            country=country,
+            language=language,
+        ),
+        timeout=DEFAULT_TIMEOUT,
+    )
+
+    out = res.json()
 
-    url = urljoin(auth_base, 'login/sign_in')
-    query = urlencode({
-        'country': country,
-        'language': language,
-        'svcCode': SVC_CODE,
-        'authSvr': 'oauth2',
-        'client_id': CLIENT_ID,
-        'division': 'ha',
-        'grant_type': 'password',
-    })
-    return '{}?{}'.format(url, query)
+    msg = out.get(DATA_ROOT)
+    if not msg:
+        raise APIError("-1", out)
+
+    if "returnCd" in msg:
+        code = msg["returnCd"]
+        if code != "0000":
+            message = msg["returnMsg"]
+            if code in API2_ERRORS:
+                raise API2_ERRORS[code]()
+            raise APIError(code, message)
```

Comment:
I don't love that this stuff has to be duplicated for the get and post queries. Maybe we can think of a creative way to avoid that.

Reply:
The 2 result are not exactly the same. One have "returnCd" and the other "returnCode"